src/lib/db/types.ts

```diff
@@ -0,0 +1,170 @@
+import { Knex } from 'knex'
+import { z } from 'zod'
+
+const char64Parser = z.string().max(64)
+
+export const tablesList = [
+  'demand',
+  'transaction',
+  'match2',
+  'processed_blocks',
+  'unprocessed_blocks',
+  'demand_comment',
+  'match2_comment',
+] as const
+
+const insertDemand = z.object({
+  owner: z.string(),
+  state: z.union([z.literal('pending'), z.literal('created'), z.literal('allocated'), z.literal('cancelled')]),
+  subtype: z.union([z.literal('demand_a'), z.literal('demand_b')]),
+  parameters_attachment_id: z.string(),
+  latest_token_id: z.union([z.number(), z.null()]),
+  original_token_id: z.union([z.number(), z.null()]),
+})
+
+const insertTransaction = z.object({
+  local_id: z.string(),
+  api_type: z.union([z.literal('match2'), z.literal('demand_a'), z.literal('demand_b')]),
+  state: z.union([z.literal('submitted'), z.literal('inBlock'), z.literal('finalised'), z.literal('failed')]),
+  transaction_type: z.union([
+    z.literal('creation'),
+    z.literal('proposal'),
+    z.literal('accept'),
+    z.literal('comment'),
+    z.literal('rejection'),
+    z.literal('cancellation'),
+  ]),
+  hash: char64Parser,
+})
+
+const insertMatch2 = z.object({
+  optimiser: z.string().max(48),
+  member_a: z.string().max(48),
+  member_b: z.string().max(48),
+  state: z.union([
```

Comment:
could some of the db types be inferred and replace types in `src/models`? I'm thinking types like `Match2State`